docs: introduce custom version selector in api.html

not-my-profile opened the pull request:
Resolves #27277.

#### Does this PR introduce a user-facing change?

<!--
If no, just write `None` in the release-note block below. If yes, a release note
is required: Enter your extended release note in the block below. If the PR
requires additional action from users switching to the new release, include the
string "action required".

For more information on release notes, please follow the Kubernetes model:
https://git.k8s.io/community/contributors/guide/release-notes.md
-->

```release-note
Updated the REST API documentation page to hide the Read the Docs version selector (which didn't affect the API version) and introduced a new version selector for the API version.
```

Honny1 commented on docs/source/_static/api.html:
```diff
@@ -15,25 +15,64 @@
+        oldContainer.parentNode.replaceChild(redocContainer, oldContainer);
 
-      document.body.appendChild(redoc);
+        Redoc.init(""https://storage.googleapis.com/libpod-master-releases/swagger-" + versionSelect.value + ".yaml", {
```
`Uncaught SyntaxError: missing ) after argument list` (`api.html:67:22`)

not-my-profile commented:
I agree that the outline I added looks bad, I just removed it. But I don't think styling it to be dark makes sense because I just noticed that for narrow browser widths ReDoc puts a white menu button next to it (and I adjusted the position to be better aligned with that other button):

<img width="114" height="138" alt="image" src="https://github.com/user-attachments/assets/e123b86f-a836-486e-8e04-6790c04d30bc" />

flouthoc reviewed (approved):
LGTM

mheon commented:
@ashley-cui Mind doing final review?

baude commented:
/approve
